x-pack/test/api_integration/apis/ml/data_frame_analytics/update.ts

```diff
@@ -0,0 +1,275 @@
+        expect(fetchedJob.max_num_threads).to.eql(commonJobConfig.max_num_threads);
+      });
+
+      it('should only update model_memory_limit field of analytics job when model_memory_limit is sent in request', async () => {
+        const analyticsId = `${jobId}_3`;
+
+        const requestBody = {
+          model_memory_limit: '61mb',
+        };
+
+        const { body } = await supertest
+          .post(`/api/ml/data_frame/analytics/${analyticsId}/_update`)
+          .auth(USER.ML_POWERUSER, ml.securityCommon.getPasswordForUser(USER.ML_POWERUSER))
+          .set(COMMON_REQUEST_HEADERS)
+          .send(requestBody)
+          .expect(200);
+
+        expect(body).not.to.be(undefined);
+
+        const fetchedJob = await getDFAJob(analyticsId);
+
+        expect(fetchedJob.model_memory_limit).to.eql(requestBody.model_memory_limit);
+        expect(fetchedJob.allow_lazy_start).to.eql(commonJobConfig.allow_lazy_start);
+        expect(fetchedJob.description).to.eql(testJobConfigs[3].description);
+        expect(fetchedJob.max_num_threads).to.eql(commonJobConfig.max_num_threads);
+      });
+
+      it('should only update max_num_threads field of analytics job when max_num_threads is sent in request', async () => {
+        const analyticsId = `${jobId}_4`;
+
+        const requestBody = {
+          max_num_threads: 2,
+        };
+
+        const { body } = await supertest
+          .post(`/api/ml/data_frame/analytics/${analyticsId}/_update`)
+          .auth(USER.ML_POWERUSER, ml.securityCommon.getPasswordForUser(USER.ML_POWERUSER))
+          .set(COMMON_REQUEST_HEADERS)
+          .send(requestBody)
+          .expect(200);
+
+        expect(body).not.to.be(undefined);
+
+        const fetchedJob = await getDFAJob(analyticsId);
+
+        expect(fetchedJob.max_num_threads).to.eql(requestBody.max_num_threads);
+        expect(fetchedJob.model_memory_limit).to.eql(commonJobConfig.model_memory_limit);
+        expect(fetchedJob.allow_lazy_start).to.eql(commonJobConfig.allow_lazy_start);
+        expect(fetchedJob.description).to.eql(testJobConfigs[4].description);
+      });
+
+      it('should not allow to update analytics job for unauthorized user', async () => {
+        const analyticsId = `${jobId}_0`;
+        const requestBody = {
+          description: 'Unauthorized',
+        };
+
+        const { body } = await supertest
+          .post(`/api/ml/data_frame/analytics/${analyticsId}/_update`)
+          .auth(USER.ML_UNAUTHORIZED, ml.securityCommon.getPasswordForUser(USER.ML_UNAUTHORIZED))
+          .set(COMMON_REQUEST_HEADERS)
+          .send(requestBody)
+          .expect(404);
+
+        expect(body.error).to.eql('Not Found');
+        expect(body.message).to.eql('Not Found');
+
+        const fetchedJob = await getDFAJob(analyticsId);
+        // Description should not have changed
+        expect(fetchedJob.description).to.eql(editedDescription);
```

Comment:
At first glance it would be hard to know `fetchedJob.description` should equal `editedDescription` based on the variable name, even though I know we already made a request to update the job to that editedDescription earlier. I wonder if it would be clearer to `expect description to not eql requestBody.description`

Reply:
I think I prefer to have an explicit check for the description so we know it hasn't been changed in any way. A not equal check could potentially miss any undesired changes to the description.

Reply:
I agree with @alvarezmelissa87 - we should stay with the explicit checks. The only thing that comes to my mind for making it more clear, is to maybe rename `fetchedJob` to `updatedJob`?